### PR TITLE
Refactor: Workspace factory controller (3/3)

### PR DESCRIPTION
**Pull request by chynu**
Refactored 16 functions from `wfactory_controller.js` into new refactor files. Empty functions, not implemented. Made only minor edits to JSdoc.

Will add function contents, then add new features once refactor is finished.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-devtools/75)

### CODE REVIEW

**Comment by rachel-fenichel** [comment]
Reviewed 3 of 6 files at r1.
Review status: 3 of 6 files reviewed at latest revision, 6 unresolved discussions.

---

*[src/factory_utils.js, line 1070 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJUKkmzcTl2OQmzkrN:-KoJUKkmzcTl2OQmzkrO:bvhs5pq) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/factory_utils.js#L1070)):*
> ```JavaScript
>  * category (case insensitive).
>  *
>  * @param {string} name The name of the category that should be checked if it's
> ```

"The name of the category that should be checked if it's in StandardCategories categoryMap" feels awkward.  Let's do

"The category name to be checked."

---

*[src/factory_utils.js, line 1076 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJXkXD32zLh77ldO6w:-KoJXkXD32zLh77ldO6x:b8bn2y7) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/factory_utils.js#L1076)):*
>   for (var category in StandardCategories.categoryMap) {
>     if (name.toLowerCase() == category) {
>       return true;
>     }
>   }
>   return false;

I think you can do
`return name.toLowerCase() in StandardCategories.categoryMap;`

---

*[src/controller/workspace_controller.js, line 58 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJUw_TdIQI9sasnwwR:-KoJUw_UD82af_OUz7KW:behehgw) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/controller/workspace_controller.js#L58)):*
> ```JavaScript
> 
>   /**
>    * Saves the WorkspaceContents currently being edited into Project object.
> ```

[...] into the current Project.

---

*[src/view/toolbox_editor_view.js, line 151 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJVtvtiFep0UTVYqPo:-KoJVtvtiFep0UTVYqPp:bk7n0h9) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/view/toolbox_editor_view.js#L151)):*
> ```JavaScript
> 
>   /**
>    * Changes the color of the selected category. Return if selected element is
> ```

"Return if selected element is a separator" is hard to parse.  Do you mean "Does nothing if the selected element is a separator."?

---

*[src/view/toolbox_editor_view.js, line 154 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJWKzNL4mc0eEP0Cek:-KoJWKzNL4mc0eEP0Cel:b5536qg) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/view/toolbox_editor_view.js#L154)):*
> ```JavaScript
>    * a separator.
>    * @param {string} color The color to change the selected category. Must be
>    * a valid CSS string.
> ```

Indent +4

---

*[src/view/toolbox_editor_view.js, line 185 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJWUl-Y_8VLD_xDLdi:-KoJWUl-Y_8VLD_xDLdj:bmk4tdl) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/view/toolbox_editor_view.js#L185)):*
> ```JavaScript
>   /**
>    * Loads a Standard Category by name and switches to it. Leverages
>    * StandardCategories. Returns if cannot load standard category.
> ```

Similar question here about "Return if".  Does it mean "Does nothing if" ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/75)*

**Comment by rachel-fenichel** [comment]
Reviewed 3 of 6 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/75)*

**Comment by chynu** [comment]
Review status: 3 of 6 files reviewed at latest revision, 6 unresolved discussions.

---

*[src/factory_utils.js, line 1070 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJUKkmzcTl2OQmzkrN:-KoJZ-ss7bRoh9apHa4X:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/factory_utils.js#L1070)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

"The name of the category that should be checked if it's in StandardCategories categoryMap" feels awkward.  Let's do

"The category name to be checked."
</blockquote></details>

Done.

---

*[src/factory_utils.js, line 1076 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJXkXD32zLh77ldO6w:-KoJZIpHPNWKCMLaRlWw:bsuwzsl) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/factory_utils.js#L1076)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

>   for (var category in StandardCategories.categoryMap) {
>     if (name.toLowerCase() == category) {
>       return true;
>     }
>   }
>   return false;

I think you can do
`return name.toLowerCase() in StandardCategories.categoryMap;`
</blockquote></details>

ooh, done.

---

*[src/controller/workspace_controller.js, line 58 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJUw_TdIQI9sasnwwR:-KoJZpObDy6A_10KRZTO:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/controller/workspace_controller.js#L58)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

[...] into the current Project.
</blockquote></details>

Done.

---

*[src/view/toolbox_editor_view.js, line 151 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJVtvtiFep0UTVYqPo:-KoJ_jwCm8GGH2-TsWGe:bzd56fx) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/view/toolbox_editor_view.js#L151)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

"Return if selected element is a separator" is hard to parse.  Do you mean "Does nothing if the selected element is a separator."?
</blockquote></details>

Done. I should've proofread the written JSdoc more thoroughly, oops.

---

*[src/view/toolbox_editor_view.js, line 154 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJWKzNL4mc0eEP0Cek:-KoJ_ztUw44sIHu8CisH:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/view/toolbox_editor_view.js#L154)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Indent +4
</blockquote></details>

Done.

---

*[src/view/toolbox_editor_view.js, line 185 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJWUl-Y_8VLD_xDLdi:-KoJa4qc6qOXS9zG_wtk:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/e9126a63d4e636b71a9152d3762825be70204acb/src/view/toolbox_editor_view.js#L185)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Similar question here about "Return if".  Does it mean "Does nothing if" ?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/75)*

**Comment by rachel-fenichel** [comment]
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/factory_utils.js, line 1074 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/75#-KoJcPNzy2fw8UKiWuyb:-KoJcNsDb333D_5uUyoZ:b6kgzk2) ([raw file](https://github.com/google/blockly-devtools/blob/d2c34029dae0adc7b048bd7b070d930661c768cd/src/factory_utils.js#L1074)):*
> ```JavaScript
>  */
> FactoryUtils.isStandardCategoryName = function(name) {
>   return name.toLowerCase() in StandardCategories.categoryMap
> ```

Missing semicolon at end.

Please double-check that this does what you want--checking if an item is in a map can be subtle.  

The easiest way to double-check is to open up chrome developer tools from Blockly DevTools and call this function with a variety of inputs.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/75)*

**Comment by rachel-fenichel** [comment]
:lgtm:

---

Review status: 5 of 8 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/75#-:-KoJj0eYhDIrzfyz3j7C:bnfp4nl)*
